docs/reference_guide.md

```diff
@@ -234,7 +234,7 @@ Examples in situ:
 
 ### 1. bpf_probe_read()
 
-Syntax: ```int bpf_probe_read(void *dst, int size, void *src)```
+Syntax: ```int bpf_probe_read(void *dst, int size, const void *src)```
```

Comment:
The below is kernel uapi/linux/bpf.h definitions:

linux/bpf.h: * int bpf_probe_read(void *dst, int size, void *src)
linux/bpf.h: * int bpf_probe_read_str(void *dst, int size, const void *unsafe_ptr)

I know it is a little bit annoying so it is better to keep the func signature in line with
the kernel tree.

I will check whether we could add "const" modifier to "void *src" in kernel:include/uapi/linux/bpf.h.

Reply:
Thanks for feedback. BPF is new to me and I too noticed after sending the patch that part of the changes should go first to the linux.git tree as `src/cc/compat/linux/bpf.h` comes copied from there.

Please let me know the outcome of your study. (imho it is ok and should be done there and not only for 2 functions touched in my patch but for others as well - e.g. for `bpf_probe_write_user()` for `void *src` too).